src/layouts/PageLayout.tsx

```diff
@@ -1,36 +1,24 @@
-import React, { useState, useRef } from "react";
+import React, { useState, useRef, useEffect, RefObject } from "react";
+import { Outlet, useOutletContext, useParams } from "react-router-dom";
 import Box from "@mui/material/Box";
 import { Typography } from "@mui/material";
 import { Popup } from "../components";
 import { Environments } from "../features/environments";
-import { EnvironmentCreate } from "../features/environmentCreate";
-import { EnvironmentDetails } from "../features/environmentDetails";
-import { PageTabs } from "../features/tabs";
 import { StyledScrollContainer } from "../styles";
-import { useAppSelector } from "../hooks";
-
-interface IUpdateEnvironment {
-  data: {
-    show: boolean;
-    description: string;
-  };
-}
+import { useAppDispatch, useAppSelector } from "../hooks";
+import { closeNotification } from "../features/notification/notificationSlice";
 
 export const PageLayout = () => {
-  const { selectedEnvironment, newEnvironment } = useAppSelector(
-    state => state.tabs
-  );
+  const dispatch = useAppDispatch();
   const [refreshEnvironments, setRefreshEnvironments] = useState(false);
-  const [notification, setNotification] = useState({
-    show: false,
-    description: ""
-  });
-  const scrollRef = useRef<HTMLDivElement | null>(null);
+  const notification = useAppSelector(state => state.notification);
+  const scrollRef = useRef<HTMLDivElement>(null);
 
-  const onUpdateOrCreateEnv = ({ data }: IUpdateEnvironment) => {
```

Comment:
I decided to get rid of this callback function because:

- Turning the child components into routed components means that the parent no longer imports the child components and so cannot pass properties directly to them (see line 47 on the right for how a route parent has to pass properties to route children)
- It seemed that having a centralized place in the app state for notifications is a good idea. PageLayout probably shouldn't own these functions.
- Keeping the `setRefreshEnvironment` behavior would be easy with the useEffect hook. However this will need to be changed in the future (I will add a TODO comment)